x-pack/plugins/beats_management/public/lib/adapters/framework/kibana_framework_adapter.ts

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/* eslint-disable max-classes-per-file */
+import { IScope } from 'angular';
+import { PathReporter } from 'io-ts/lib/PathReporter';
+import { isLeft } from 'fp-ts/lib/Either';
+import { first } from 'rxjs/operators';
+import { SecurityPluginSetup } from '../../../../../security/public';
+import { BufferedKibanaServiceCall, KibanaAdapterServiceRefs, KibanaUIConfig } from '../../types';
+import {
+  FrameworkAdapter,
+  FrameworkInfo,
+  FrameworkUser,
+  RuntimeFrameworkInfo,
+  RuntimeFrameworkUser,
+} from './adapter_types';
+import {
+  ManagementSetup,
+  RegisterManagementAppArgs,
+} from '../../../../../../../src/plugins/management/public';
+import { LicensingPluginSetup } from '../../../../../licensing/public';
+import { BeatsManagementConfigType } from '../../../../common';
+
+export class KibanaFrameworkAdapter implements FrameworkAdapter {
+  public get info() {
+    if (this.xpackInfo) {
+      return this.xpackInfo;
+    } else {
+      throw new Error('framework adapter must have init called before anything else');
+    }
+  }
+
+  public get currentUser() {
+    return this.shieldUser!;
+  }
+  private xpackInfo: FrameworkInfo | null = null;
+  private adapterService: KibanaAdapterServiceProvider;
+  private shieldUser: FrameworkUser | null = null;
+  constructor(
+    private readonly PLUGIN_ID: string,
+    private readonly management: ManagementSetup,
+    private readonly getBasePath: () => string,
+    private readonly licensing: LicensingPluginSetup,
+    private readonly securitySetup: SecurityPluginSetup | undefined,
+    private readonly config: BeatsManagementConfigType,
+    public readonly version: string
+  ) {
+    this.adapterService = new KibanaAdapterServiceProvider();
+  }
+
+  public setUISettings = (key: string, value: any) => {
+    this.adapterService.callOrBuffer(({ config }) => {
```

Comment:
Is it a bridge to the legacy angular code?

Reply:
Not sure, I didn't touch this (just moved). I'll double check though.

Reply:
Honestly it looks like this code is just broken (and probably has been for some time). The adapter service buffers calls until its `$get` method is called, which I cannot find any place where that is called in the legacy implementation.

Not sure if worth fixing this now or not. I'm just too afraid of breaking things here without any tests.